physics/tools/canopy_utils_mod.f

```diff
@@ -0,0 +1,52 @@
+       module canopy_utils_mod
+
+        implicit none
+
+
+       contains
+
+!--------------------------------------------------------------------------
+
+       function IntegrateTrapezoid(x, y)
+           !! Calculates the integral of an array y with respect to x
+           !using the trapezoid
+           !! approximation. Note that the mesh spacing of x does not
+           !have to be uniform.
+           real, intent(in)  :: x(:)                !! Variable x
+           real, intent(in)  :: y(size(x))          !! Function y(x)
+           real              :: IntegrateTrapezoid  !! Integral of y(x)dx
+       ! Integrate using the trapezoidal rule
+           associate(n => size(x))
+             IntegrateTrapezoid = sum((y(1+1:n-0) + y(1+0:n-1))*
+     &                            (x(1+1:n-0) - x(1+0:n-1)))/2
+           end associate
+       end function
+
+! ---------------------------------------------------------------------------
+
+      function interp_linear1_internal(x,y,xout) result(yout)
+        !! Interpolates for the y value at the desired x value,
+        !! given x and y values around the desired point.
+
+        implicit none
+
+        real, intent(IN)  :: x(2), y(2), xout
+        real :: yout
+        real :: alph
+
+        if ( xout .lt. x(1) .or. xout .gt. x(2) ) then
+            write(*,*) "interp1: xout < x0 or xout > x1 !"
+            write(*,*) "xout = ",xout
+            write(*,*) "x0   = ",x(1)
+            write(*,*) "x1   = ",x(2)
+            stop
```

Comment:
I don't see this function actually being used in the CCPP physics, but if it were, stopping the model is not allowed in a CCPP physics scheme. To do this correctly, you would need to pass in the CCPP error handling variables (`ccpp_error_message` and `ccpp_error_code`) to this function and set them and return them to the calling routine (which should also return them) so that the host can shut things down in a more controlled way.

Reply:
@grantfirl You are correct and here we do not use this, so we will remove the entire function from the module fixing this issue.  @iri01